### PR TITLE
[move-ide] Split symbolication implementation into smaller functions

### PR DESCRIPTION
**Pull request by awelc**
## Description 

This PR contains a refactoring to split a super-long `get_symbols` function into two smaller ones.

It not only makes code mode readable but also enables a more optimized implementation of the test suite. Previously, all symbols had to be recomputed when tests involved a cursor, even though it was only cursor that was changing, rather than symbols themselves. With this PR, we can recompute the cursor only, which brings the time to run all `move-analyzer` tests from minutes to (tens of) seconds.

## Test plan 

All existing tests must pass

### CODE REVIEW

**Review by cgswords — COMMENTED** [review]
LGTM, nice little bit of cleanup and refactoring.

**Review by cgswords — APPROVED** [review]
LGTM, nice little bit of cleanup and refactoring.

File: external-crates/move/crates/move-analyzer/tests/ide_testsuite.rs [hunk 1]
```diff
@@ -189,13 +189,25 @@ impl CompletionTest {
...
+        // but we need to recompute the cursor each time
+        let cursor_path = use_file_path.to_path_buf();
+        let cursor_info = Some((&cursor_path, use_pos));
+        let mut symbols_computation_data = SymbolsComputationData::new();
+        let mut cursor_context = compute_symbols_pre_process(
+            &mut symbols_computation_data,
+            &compiled_pkg_info,
+            cursor_info,
         );
+        cursor_context = compute_symbols_parsed_program(
```
Comment (cgswords) [inline external-crates/move/crates/move-analyzer/tests/ide_testsuite.rs:203]: Consider adding a comment that this skips typed computation for speed

Comment (awelc) [reply external-crates/move/crates/move-analyzer/tests/ide_testsuite.rs:205]: Thank you! Done!

**Pull request merged**